pkg/api/norman/store/nodetemplate/store.go

```diff
@@ -215,11 +215,22 @@ func (s *nodeTemplateStore) replaceCloudCredFields(apiContext *types.APIContext,
 	var fields []string
 	for key := range cred.Data {
 		splitKey := strings.SplitN(key, "-", 2)
-		if len(splitKey) == 2 && splitKey[0] == credConfigName {
+		if len(splitKey) == 2 && splitKey[0] == credConfigName && !needToBePreserved(credConfigName, splitKey[1]) {
 			delete(toReplace, splitKey[1])
 			fields = append(fields, splitKey[1])
 		}
 	}
 	logrus.Debugf("replaceCloudCredFields: %v for credID %s", fields, credID)
 	return nil
 }
+
+// Some fields in credential need to be saved in node template, and these fields do not affect security
+// Parameters:
+// credConfigName - Credential config name.
+// key - Fields that need to be preserved
+func needToBePreserved(credConfigName, key string) bool {
```

Comment:
I don't think we need this filter here. Should keep removing all credential fields.

Reply:
Ok, I removed this code and finished confirming with the front end. Let the front end get the `environment` from azure credential uniformly.